### ISSUE TITLE
Stuck after a few hours

### ISSUE THREAD

**Issue by highstrike** [lloesche/valheim-server-docker#24]
Running my server on a windows machine with docker for windows

My game world gets stuck after a few hours of playing and it kicks me off with a disconnect message, Afterwhich I can connect to the server, but it asks for no password (its password protected) and all I see is a black screen, not even the loading icon.

I have to restart the docker container for it to "work". And I say "work" because after the restart, everything I've ever built is gone and I have to grab a backup and use that.

I'm not sure why this is happening, either my world is corrupt or my character is.

Anyone else having this issue?

**Comment by lloesche** [comment]
I've not noticed that particular behavior. What we have seen on our server though is that after a while the server gets really laggy and desyncs a lot. Actions like opening a door or hitting an enemy register seconds later. And then there is a strange queuing behavior where I once saw in-game actions that my brother did a good 3-4 minutes after they happened. As if the server doesn't skip any frames or has a really high buffer or something like that. In all cases restarting the server solved the issue leading me to believe that there is some sort of resource leak going on.

As Docker on Windows is really just Docker in a Linux VM running on Windows I would suspect that you are running out of VM resources. Like, if I can feel the effects on a Linux server that's got 64GB of Ram and a 6 Core Xeon CPU after a couple of hours runtime I would assume that a resource leak is all the more pronounced in an environment that's heavily CPU and memory constrained by a Hypervisor.

To debug this further and if you have the resources to spare I would allow Docker on Windows to consume more CPU Cores and memory. Like if the VM is currently using e.g. 2 Cores and 2 GB maybe increase it to 4 Cores and 8 GB. Just to see if that changes anything.

That said on Windows it's likely a better idea to not run the Linux version of the server at all and instead opt for the native Windows binary. Albeit not running in a Docker container.

**Comment by highstrike** [comment]
You're right, I'm on Docker for Windows with Linux containers. I have increased it to 6 cores and 8gb of ram and also moved the volume on my nvme ssd.

I haven't noticed the laggy behaviour but I did notice the high resource leak, maybe the dedicated server needs some more work 😅 

Since descovering docker and containerizing everything I will never go back to installing stuff directly on the host os, especially now that I know the world files get corrupt, without your backup mechanism I'm toast.

Thanks for the suggestions!
I've played for 8 hours straight and no issue so far
